### DISCUSSION TITLE
Cannot Reference Default Value Of A Text Field With Another Text Field On Page Load

### DISCUSSION THREAD

**Discussion by CosmasSC128** [Budibase/budibase#6032] [Help] [answered]
I have a Text Field A with default value X, and I want Field B with default value to be X + Y, that updates as X is changed.

On change functions enable this, but when the page loads, Field A fails to pass its default value of X to Field B. This value is only passed once Field A has a focus change.

Is this a bug or an intended feature?

**Comment by melohagan** [comment, marked as answer]
Hey @CosmasSC128 

This is intended, to avoid excessive data reloading and rendering of the page.

**Comment by CosmasSC128** [comment]
I have created an image design preview app function. Imagine base products, and you want to add a banner Sale! -20% over the item at the top. It's possible to have styling changes to this app adjust as fields are edited by a human, but on page load, the preview is broken/does not work.

Can there be no feature change, or way of implementing the default values of the fields are loaded immediately? Not in a setting, etc?

It removes a lot of dynamic useful behaviour to never have that open, rather than most of the time not have that occur. Sorry if this is obvious, but where does one make an official feature request?

**Reply by melohagan** [reply]
Hey @CosmasSC128 

You can create a new feature request here: https://github.com/Budibase/budibase/issues/new/choose
